src/core/doall/src/DOALL_chunking.cpp

```diff
@@ -177,14 +185,68 @@ void DOALL::rewireLoopToIterateChunks(LoopContent *LDI, DOALLTask *task) {
     assert(taskLoopValue != nullptr);
 
-    /*
+    // create code to check if the PVSCC must chunkstep
+    auto isChunkCompleted =
+        cast<SelectInst>(chunkPHI->getIncomingValueForBlock(taskLoopBlock))
+            ->getCondition();
+
+    // set builder insert points
+    IRBuilder<> headerBuilder(task->getCloneOfOriginalBasicBlock(
+        LDI->getLoopStructure()->getHeader()));
+    headerBuilder.SetInsertPoint(&*headerBuilder.GetInsertBlock()->begin());
+    IRBuilder<> latchBuilder(task->getCloneOfOriginalBasicBlock(
+        *LDI->getLoopStructure()->getLatches().begin()));
+    latchBuilder.SetInsertPoint(latchBuilder.GetInsertBlock()->getTerminator());
+
+    // PVSCC will use the absolute iteration as part of our handling for them:
+    // get the counter for that or inject it
+    if (getOrInjectIterCounter == nullptr) {
+      /*
+       * Determine value of the start of this core's next chunk
+       * from the beginning of the next core's chunk.
+       * Formula: (next_chunk_initialValue + (step_size * (num_cores - 1) *
+       * chunk_size)) % period
+       */
+
+      // acquire the info needed to chunkstep *the absolute iteration counter*
+      auto onesValueForChunking = ConstantInt::get(chunkCounterType, 1);
+      auto coreIDxChunkSize = entryBuilder.CreateMul(task->taskInstanceID,
+                                                     task->chunkSizeArg,
+                                                     "coreIdx_X_chunkSize");
+      auto numCoresMinus1 = entryBuilder.CreateSub(task->numTaskInstances,
+                                                   onesValueForChunking,
+                                                   "numCoresMinus1");
+      auto chunkStepSize = entryBuilder.CreateMul(numCoresMinus1,
+                                                  task->chunkSizeArg,
+                                                  "numCoresMinus1_X_chunkSize");
+
+      // build absolute iteration counter
+      getOrInjectIterCounter = headerBuilder.CreatePHI(
+          llvm::Type::getInt64Ty(headerBuilder.getContext()),
+          2,
+          "iterCounter");
+      getOrInjectIterCounter->addIncoming(coreIDxChunkSize, preheaderClone);
+      auto iterIncrement =
+          latchBuilder.CreateAdd(getOrInjectIterCounter, onesValueForChunking);
+      auto iterChunkStep = latchBuilder.CreateAdd(iterIncrement, chunkStepSize);
+      auto iterSelect = latchBuilder.CreateSelect(isChunkCompleted,
+                                                  iterChunkStep,
+                                                  iterIncrement);
+      getOrInjectIterCounter->addIncoming(iterSelect,
+                                          latchBuilder.GetInsertBlock());
+    }
+
+    assert((getOrInjectIterCounter != nullptr)
+           && "DOALL_chunking: no iterCounter\n");
+
+    /* DD: we choose to not do this and instead replace the whole PVSCC with a
```

Comment:
Please remove "DD"

Reply:
Also, please follow the comment pattern we have been using throughout the codebase